### DISCUSSION TITLE
Allow same user account with different Identity Providers

### DISCUSSION THREAD

**Discussion by cesdperez** [keycloak/keycloak#17532] [Q&A]
Hi!

I'm using Keycloak as an Identity Broker, and I have multiple Identity Providers set up.

In my setup, it could be that a user exists in multiple of those Identity Providers. Moreover, the user email is used as username in many Identity Providers, and that results in a clash and a message from Keycloak "User with username xxx@xxx.com already exists. Please login to account management to link the account."

What I'd like to achieve is that the same (existing) user account is used in Keycloak, and that happens transparently (without user interaction). To me, it doesn't matter from which Identity Provider did the user come from.

I tried different combinations of First broker login Authentication flows, but I couldn't achieve that.
I also tried allowing duplicated emails in the realm settings, but that doesn't solve the issue as what actually happens are duplicated usernames.

Is there a way to achieve this?
Thanks for your help.

Keycloak version: 20.0.1

**Comment by dhtek** [comment]
Did you find a solution ? We would like to do the same...
